x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/ProfilingPlugin.java

```diff
@@ -55,14 +55,22 @@ public class ProfilingPlugin extends Plugin implements ActionPlugin {
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
+
+    // *Internal* setting meant as an escape hatch if we need to skip the check for outdated indices for some reason.
+    public static final Setting<Boolean> PROFILING_CHECK_OUTDATED_INDICES = Setting.boolSetting(
```

Comment:
This is intentionally an undocumented setting. I only added this to give us an option to circumvent the check if we should need this in a support case or to debug an issue. It is not meant to be changed without our explicit guidance.